### PR TITLE
snyk: Fix fingerprint in audit_logs.

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
```
Fix fingerprint in audit_logs.

Certain group events in audit_logs don't contain `org_id`.
Example: `event: group.service_account.create`.

   - Update fingerprint processor with ignore_missing: true.
   - Add group_id to fingerprint processor.

```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

`eval "$(elastic-package stack shellinit)" && elastic-package test pipeline --generate -v`

Pipeline tests before (error):
```
[0] unexpected pipeline error: [missing field [json.org_id] when calculating fingerprint]
```

Pipeline tests after (success):
```
--- Test results for package: snyk - START ---
╭─────────┬─────────────────┬───────────┬──────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE │ DATA STREAM     │ TEST TYPE │ TEST NAME                                                │ RESULT │ TIME ELAPSED │
├─────────┼─────────────────┼───────────┼──────────────────────────────────────────────────────────┼────────┼──────────────┤
│ snyk    │ audit           │ pipeline  │ (ingest pipeline warnings test-snyk-audit.log)           │ PASS   │  458.18425ms │
│ snyk    │ audit           │ pipeline  │ test-snyk-audit.log                                      │ PASS   │  50.105625ms │
│ snyk    │ audit_logs      │ pipeline  │ (ingest pipeline warnings test-snyk-audit.json)          │ PASS   │ 436.509333ms │
│ snyk    │ audit_logs      │ pipeline  │ test-snyk-audit.json                                     │ PASS   │ 103.105958ms │
│ snyk    │ issues          │ pipeline  │ (ingest pipeline warnings test-snyk-issues.json)         │ PASS   │ 448.944292ms │
│ snyk    │ issues          │ pipeline  │ test-snyk-issues.json                                    │ PASS   │     76.362ms │
│ snyk    │ vulnerabilities │ pipeline  │ (ingest pipeline warnings test-snyk-vulnerabilities.log) │ PASS   │ 466.026375ms │
│ snyk    │ vulnerabilities │ pipeline  │ test-snyk-vulnerabilities.log                            │ PASS   │  81.281042ms │
╰─────────┴─────────────────┴───────────┴──────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: snyk - END   ---
Done
```

### CODE REVIEW

File: packages/snyk/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.25.1"
+  changes:
+    - description: Fix fingerprint in audit_logs.
+      type: bugfix
+      link: https://github.com/elastic/integrations/pull/1111
```
Comment (ShourieG) [inline packages/snyk/changelog.yml:6]: can we update it with the proper PR number.

**Comment by efd6** [comment]
In commit message:
```
Certain group events in audit_logs don't contain `org_id`.
```

File: packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -12,24 +12,31 @@ processors:
   - json:
       field: event.original
       target_field: json
+# Cleanup equivalent names
+  - rename:
+      field: json.groupId
```
Comment (ShourieG) [inline packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml:17]: just curious, why are we renaming here, doesn't the source log already contain "group_id" ? I didn't see any **groupId** in the original log.

File: packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -12,24 +12,31 @@ processors:
...
+      target_field: json.group_id
+      ignore_missing: true
+  - rename:
+      field: json.orgId
+      target_field: json.org_id
+      ignore_missing: true
   - fingerprint:
       fields:
         - json.org_id
+        - json.group_id
```
Comment (ShourieG) [inline packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml:27]: if org_id is absent, is group_id always gauranteed to be present ? What if they are both present or conditionally present, will it not impact fingerprints thus creating duplicates ?

File: packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (kcreddy) [reply packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml:17]: This has been added earlier and I just moved the block to add onto the fingerprint correctly.

https://github.com/elastic/integrations/blob/d4c56155dd8d227dd75ffed12d494952d29071b7/packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml#L25-L28

File: packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (kcreddy) [reply packages/snyk/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml:27]: Groups contains organization(s). So, group_id is always present as per the API.

**Review by ShourieG — APPROVED** [review]
LGTM

**Pull request merged**